Add better name resolution for homematic channels

weissm opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
Adds better namining resolution for channels by modifying the len variable for later resolution in homeassistant/components/homematic/__init__.py:_create_ha_id.

- Example: 
  - before the channel numbers are used as name:
![grafik](https://user-images.githubusercontent.com/4221212/95651690-589a5100-0aec-11eb-9d5f-b4dc6963de37.png)

  - now: Names provide in CCU3 are used:
![grafik](https://user-images.githubusercontent.com/4221212/95651453-cba2c800-0aea-11eb-97e6-8a197939c3d4.png)

- Remarks: 
  - Will leverage change commited @ pyhomematic #341.
  - This is a follow-up on the request https://github.com/home-assistant/core/pull/41587. 
  - Code reviews/tests have been conducted. <unique_id> have **not** changed. 
  - Only change is here on <channel_count> clarified but minimize adding new line breaks.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [x] Breaking change (fix/feature causing existing functionality to break)
## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

weissm committed:
Added better name resolution for channels

pvizeli commented on homeassistant/components/homematic/__init__.py:
```diff
@@ -528,8 +528,13 @@ def _get_devices(hass, discovery_type, keys, interface):
             )
             for channel in channels:
+                channel_name = device.CHANNELS[channel].NAME
+                if key not in channel_name:
```
Please add a comment to explain this a bit (short)

weissm replied:
Target is to handle name resolution of channels similar to devices. In case the key itself is already the `channel_name`, the procedure will be skipped. Otherwise, the assigned "friendly" `channel_name` will be used.

before:
![grafik](https://user-images.githubusercontent.com/4221212/97802032-fcd16c80-1c40-11eb-8d42-9816cd13ba28.png)

after:
![grafik](https://user-images.githubusercontent.com/4221212/97802176-e7107700-1c41-11eb-954f-4f5eef403ba6.png)

homeassistant committed:
[ci skip] Translation update

amelchio committed:
Mock time_date sensor tests (#43141)

amelchio committed:
Fix beat calculation (#43142)

Commit:
Update xknx to 0.15.3 (#42026)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bachya committed:
Streamline SimpliSafe errors (#43117)

ludeeus committed:
Add missing 'hassio' translation string (#43127)

* Add missing 'hassio' translation string

* Fix typo...

amelchio committed:
Fix playing of Spotify URIs on Sonos (#43154)

bdraco committed:
Guard against empty ssdp locations (#43156)

balloob committed:
Update translations

balloob committed:
Further improve MFI tests (#43167)

chemelli74 committed:
Remove relative time sensor from cert_expiry (#42338)

balloob committed:
Disable parsing scientific/complex number notation in template type (#43170)

amelchio committed:
Fix time_date timestamp offsets (#43165)

balloob committed:
Improve entity domain validator (#43164)

amelchio committed:
Fix time_date interval for DST (#43166)

balloob committed:
Add test to access current request in websocket API (#43133)

doudz committed:
update zigpy_zigate to v0.7.1 (#43159)

balloob committed:
Add log message server startup (#43177)

frenck committed:
Upgrade sentry-sdk to 0.19.3 (#43176)

Danielinte committed:
Rewrite google_wifi unittest tests to pytest style (#42030)

balloob committed:
Revert opensky flight latitude and longitude (#43185)

This reverts commit 0f46916f9efec5832423ea7db8e477e34c1038d6.

balloob committed:
Prevent spider from doing I/O in the event loop (#43182)

MartinHjelmare committed:
Copy default vscode settings during bootstrap (#43180)

cgarwood committed:
Update ozw get_config_parameter websocket response (#43058)

ttuffin committed:
Hide cancelled trips from being displayed in vasttrafik integration (#43184)

balloob committed:
Fix automation in packages (#43202)

balloob committed:
Add support for selectors in services.yaml (#43162)

* Add support for selectors in services.yaml

* Add base schema validation

czechmark committed:
Update fitbit battery status  (#42980)

* Read the device_status in order to update the fitbit battery

* Read the device_status in order to update the fitbit battery

bachya committed:
Revert "Remove YAML config for Tile (#43064)" (#43199)

This reverts commit 19f48e180c7e6a34bd93d7a849147a8de491771c.

homeassistant committed:
[ci skip] Translation update

allenporter committed:
Catch the right nest stream refresh exception error (#43189)

bdraco committed:
Fix typo in lw12wifi shoud_poll (#43213)

thomasdelaet committed:
Add quarter-hour period feature for utility_meter component (#41999)

* add support for quarter-hourly intervals to utility meter (15 minutes)

* add tests

* add test for every quarter

* Update homeassistant/components/utility_meter/sensor.py

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>

cmroche committed:
Update greeclimate to 0.10.2 (#43206)

Jc2k committed:
Add initial camera support to homekit_controller (#43100)

b4dpxl committed:
Add support for Broadlink BG1 devices (#42314)

* Support for BG1 switches after config flow updates to core Broadlink component

* updates based on @felipediel feedback

* Update updater.py

* Update switch.py

* Update switch.py

Commit:
Bump elkm1-lib to 0.8.8 (#43230)

adrum committed:
Add HomeKit humidifier/dehumidifier (#42311)

* add HomeKit humidifier/dehumidifier

* added more test coverage

* simplified char logic

Co-authored-by: Quentame <polletquentin74@me.com>

* use mode constants

* Renamed HomeKit Contorller

Co-authored-by: Quentame <polletquentin74@me.com>

* improved threshold logic

* split up homekit humidifier into 2 entities

* fixed tests

* fixed mode and switch logic

* added set mode tests

* removed redundant methods present in base class

Co-authored-by: Quentame <polletquentin74@me.com>

michaeldavie committed:
Remove OpenCV dependecy from Environment Canada (#43235)

* Bump env_canada to 0.2.2

* Revert PR #38731

BKPepe committed:
Upgrade youtube_dl to version 2020.11.12 (#43231)

bdraco committed:
Switch ios to dispatching instead of polling (#43233)

andrewsayre committed:
Bump pyheos to 0.7.2 (#43239)

bdraco committed:
Bypass the slow update warning for group updates (#43209)

bdraco committed:
Eliminate doorbird switch polling (#43215)

andrewsayre committed:
Bump pysmartthings and pysmartapp (#43240)

Adminiuga committed:
Refactor ZHA light initialization (#43149)

* Refactor ZHA light initialization

* Don't do redundant attribute reads

homeassistant committed:
[ci skip] Translation update

flowolf committed:
Remove yessssms integration (#43200)

cmroche committed:
Update greeclimate to 0.10.3 (#43248)

* Update greeclimate to 0.10.3

* Device search needs to be mocked in tests

bdraco committed:
Set should_poll for zone entities (#43212)

doudz committed:
Update zigpy-zigate to 0.7.2 (#43252)

* update zigpy_zigate to v0.7.1

* bump zigpy-zigate to 0.7.2

javache committed:
Support gas meter capability for smartthings (#41310)

tim427 committed:
Add dsmr_reader telegram timestamp and device classes (#42909)

* Added Telegram timestamp

* added to 'dsmr/reading/timestamp' and 'dsmr/consumption/gas/read_at'

* Fixed import sorting

* Replaced 'kW' with ENERGY_KILO_WATT_HOUR

* Added device_class, changed unit_of_measurement with fallback on device_class

* Fixed typo

* Fixed 'black' coding format

* Removed fallback, added unit_of_measurement and CURRENCY_EURO as device_class

* Fixed newline

* Removed 'timestamp' unit_of_meassure

* Removed icons from defintions with device_class

* Updated device_classes

* Updated device_classes

* Updated device_classes

* Added 'entity_registry_enabled_default' properties

* Added 'entity_registry_enabled_default' properties, fixed typo

* MQTT discovery will be in another pull-request

bdraco committed:
Convert most esphome state updates to callbacks (#43246)

homeassistant committed:
[ci skip] Translation update

ctalkington committed:
Add tests for browse media image proxy (#43076)

* add tests for browse media image proxy

* Update test_init.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

mib1185 committed:
Apply suggestions from #42697 to synology_dsm (#43197)

* apply suggestions from #42697

* fix tests

* use MockConfigEntry for test

* use hass.config_entries.async_setup()

* disable default fixture

* rename marker to no_bypass_setup

chemelli74 committed:
Avoid creating battery sensor if Shelly device is external powered (#43243)

adriansuwala committed:
Rewrite ecobee unittest tests to pytest (#42584)

syssi committed:
Bump python-miio and construct version (#43267)

spacegaier committed:
Bump requests to 2.25.0 (#43279)

GuyKh committed:
Xiaomi Device Tracker - Move "Refreshing device list" to debug (#43276)

dependabot[bot] committed:
Bump actions/upload-artifact from v2.2.0 to v2.2.1 (#43272)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from v2.2.0 to v2.2.1.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v2.2.0...726a6dcd0199f578459862705eed35cda05af50b)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pavoni committed:
Bump pyvera to 0.3.11 (#43262)

pavoni committed:
Bump Pywemo to 0.5.3 (#43263)

balloob committed:
Automatically clean up executor as part of closing loop (#43284)

balloob committed:
Convert core tests to async (#43287)

balloob committed:
Update pytradfri to 7.0.3 (#43289)

TheLastProject committed:
Bump PyEssent to 0.14 (#43282)

michaeldavie committed:
Bump env_canada to 0.2.4, fix config validation (#43251)

emontnemery committed:
Make MQTT climate return PRESET_NONE when no preset is set (#43257)

reaper7 committed:
Fix Enigma2 available entity property (#43292)

Available property should return self.e2_box.is_offlin negation

fixes previous PR: https://github.com/home-assistant/core/pull/42407

uvjustin committed:
Remove pts adjustments in stream (#42399)

* Remove unnecessary pts adjustments

* Add comments

* Use -inf for initial last_dts to be more clear

* Use video first_pts as common adjuster in recorder

* Remove seek(0) before av.open

balloob committed:
Update pytradfri to 7.0.4 (#43297)

bramkragten committed:
Updated frontend to 20201111.1 (#43298)

Jc2k committed:
Refactor how entities are created for homekit_controller services (#43242)

bachya committed:
Bump aioguardian to 1.0.4 (#43299)

rikroe committed:
Bump bimmer_connected to 0.7.13 (#43294)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

homeassistant committed:
[ci skip] Translation update

CoMPaTech committed:
Add notification binary_sensor to Plugwise integration (#41473)

* Notifications extract from beta

* Remove info loggings

* Delete notification service

* Only notifications for right smiles

* Revert to correct logic

* Catchup with dev (mostly with ourselves from #41201)

* Remove debug logging

* Naming improvement

* Improve test quality as per codecov patch requirement

* Revert to original condition (and appropriately test)

* Fix delete_notification_service, bring tests fixtures up to 1.6.0 including notifications

* Review comment (@bouwew)

* Correct test value

* Re-apply #40108 fix after rebase sidestep

* Update tests/components/plugwise/test_init.py

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

* Add needed state to imports

* Remove separate gw unload code

* Change entry_fail approach

* Revert persistent notification part

* Revert persistent notification part - lint

* Update homeassistant/components/plugwise/binary_sensor.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/plugwise/binary_sensor.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Rework reuse of sensor in binary_sensor

* Explicit state attribute keys

* Remove tempfile

* List of notifications per severity

* Update homeassistant/components/plugwise/binary_sensor.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Chris Talkington <chris@talkingtontech.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

ctalkington committed:
Update directv to 0.4.0 (#43302)

* update directv to 0.4.0

* Update requirements_all.txt

* Update requirements_test_all.txt

raman325 committed:
Abort vizio discovery flow without unique ID (#43303)

* abort vizio discovery flow if unique ID cant be found because it means we cant connect

* add tests

* fix abort call

ttuffin committed:
Improvement to allow parsing of station ID in vasttrafik integration. Addresses #34851 (#43136)

dependabot[bot] committed:
Bump codecov/codecov-action from v1.0.14 to v1.0.15 (#43304)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.0.14 to v1.0.15.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.0.14...239febf655bba88b16ff5dea1d3135ea8663a1f9)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

MartinHjelmare committed:
Add progress translation key to hassfest (#43311)

klaasnicolaas committed:
Update cloud integration to 0.38.0 (#43314)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

emontnemery committed:
Bump hatasmota to 0.0.31 (#43319)

mLupine committed:
Add an option to template delay_on/off in template binary sensor (#43259)

mvn23 committed:
Fix kodi media_player unavailable at start (#41714)

elupus committed:
Improve Google Assistant cover assumed state handling (#43255)

* Set command only cover
* No need for override position now that we support command only
* Return empty state response for assumed state

Fixes #43178
Reverts: #23498

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Alex-Klein committed:
Add TV channel name to smartthings integration (#41729)

* Adds TV channel name to smartthings integration

* fix on attribute to not override key

homeassistant committed:
[ci skip] Translation update

bramkragten committed:
Updated frontend to 20201111.2 (#43334)

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
changed handling of entityid

weissm committed:
Merge branch 'homeassistant/improve/cover' of github.com:weissm/core into homeassistant/improve/cover

pvizeli commented:
Your branch is corurrept. Just to make it understandable, we never allow overwriting entity_id

firstof9 committed:
Tweak OZW Light discovery (#43325)

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Refactor ZHA tests to allow attribute reads during device initialization (#43357)

* Allow plugging zigpy attribute reads in tests

* Migrate ZHA tests to use new patched attribute reads

* Remove logging in tests

Adminiuga committed:
Update Zha dependencies (#43373)

Adminiuga committed:
Refactor ZHA sensor initialization (#43339)

* Refactor ZHA sensors to use cached values after restart

* Get attr from cluster, not channel

* Run cached state through formatter method

* Use cached values for div/multiplier for SmartEnergy channel

* Restore batter voltage from cache

* Refactor sensor to use cached values only

* Update tests

* Add battery sensor test

scarface-4711 committed:
Update denonavr to 0.9.6 (#43370)

raman325 committed:
Bump pyvizio to 0.1.57 (#43374)

emontnemery committed:
Bump hatasmota to 0.0.32 (#43360)

chemelli74 committed:
Add shelly installed firmware info (#43221)

thecode committed:
Support for Shelly Binary Input Sensors (#43313)

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

ludeeus committed:
Add back system_health_info to the base of lovelace (#43382)

amelchio committed:
Reword numeric_state trigger (#43367)

emontnemery committed:
Raise in base implementation of FanEntity.oscillate (#43354)

bdraco committed:
Fix homekit bridges when no name was provided (#43364)

allenporter committed:
Improve nest SDM integration error handling (#43271)

timmo001 committed:
Update ovoenergy to v1.1.11 (#43391)

adamkrol93 committed:
Bump wolf_smartset to 0.1.8 and handle server fetch error (#43351)

bramkragten committed:
Fix selector to return the selector type (#43395)

dr1rrb committed:
Add twinkly integration (#42103)

* Add twinkly integration

* Add tests for the Twinkly integration

* Update Twinkly client package to fix typo

* Remove support of configuration.yaml from Twinkly integration

* Add ability to unload Twinkly component from the UI

* Remove dead code from Twinkly

* Fix invalid error namespace in Twinkly for python 3.7

* Fix tests failing on CI

* Workaround code analysis issue

* Move twinkly client init out of entry setup so it can be re-used between entries

* Test the twinkly component initialization

* React to PR review and add few more tests

Toxblh committed:
Add new air-humidifier device CA4 with miot protocol (#39398)

* Add new air-humidifier device CA4 with miot protocol

* Update homeassistant/components/xiaomi_miio/fan.py

Co-authored-by: Teemu R. <tpr@iki.fi>

* xiomi_miio fan Standard speed for fan, removed static attributes

* xiomi_miio fan Standard speed for fan, removed static attributes

* xiomi_miio fan unnessary elif

* added mode_mapping

* Changed mode_mapping to constant also move reverse to constant

* Removerd water_level

* Return to exclude breaking changes

Co-authored-by: Teemu R. <tpr@iki.fi>

balloob committed:
Check config to use config platforms (#43407)

homeassistant committed:
[ci skip] Translation update

ehendrix23 committed:
Update pymyq to 2.0.10 (#43413)

* Bump version of pymyq to 2.0.9

* Bump version of pymyq to 2.0.10

ahayworth committed:
Update python-awair to 0.2.1 (#43415)

This version of python awair has support for the local api - which is
cool! But more importantly, this version also is less strict about the
versions of aiohttp it depends on.

This version prepares for the upcoming pip resolver changes outlined here:
https://pyfound.blogspot.com/2020/03/new-pip-resolver-to-roll-out-this-year.html#what-will-change

cc https://github.com/ahayworth/python_awair/issues/18

Bre77 committed:
Increase Advantage Air retry limit for older systems (#43417)

Commit:
Update zigpy-zigate to 0.7.3 (#43427)

* Update zigpy-zigate to 0.7.3
Fix probing pizigate

* Update zigpy-zigate to 0.7.3
Fix probing pizigate

briglx committed:
Add Srp energy component (#41091)

100ferhas committed:
Update Alexa supported languages (#43139)

elupus committed:
Support openRelativePercent for google assistant covers (#43336)

balloob committed:
Add support for checking minimum HA version (#43350)

spacegaier committed:
Provide HA username via auth WS (#43283)

balloob committed:
Fix time trigger based on entities ignoring entities if initially in the past (#43431)

balloob committed:
Verify that we register blueprints on automation setup (#43434)

RobBie1221 committed:
Update dsmr_parser to 0.23 (#43403)

alemuro committed:
Upgrade Beewi Smartclim component to 0.0.10 (#43441)

spacegaier committed:
Fix empty local_ip config flow (#43333)

springstan committed:
Use weather attribute conditions constants (#39945)

bachya committed:
Move Flo logger to a package logger (#43449)

bachya committed:
Move Ambient PWS logger to a package logger (#43448)

MartinHjelmare committed:
Move legacy device tracker setup to legacy module (#43447)

MartinHjelmare committed:
Remove salt integration for webscraping (#43452)

bachya committed:
Move Notion logger to a package logger (#43450)

scop committed:
Add ONVIF PTZ Stop support (#39734)

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Remove zigpy monkey patching (#43456)

* Use event handler for incoming messages from zigpy

* Bump up zha dependency

CoMPaTech committed:
Change Plugwise integration to plugwise module (#43036)

* Switch to plugwise module and forthcoming changes

* Adjusted according to review

* Fix leaving out domain for tests

* Add tests for exceptions

* Add more tests for exceptions

* Version bump

* Wording on test

* Catch-up with dev

FrnchFrgg committed:
Avoid arbitrarily reducing ZHA climate temperature information (#43442)

The Zigbee specification mandates that temperatures are given in
1/100ths of degrees, and several devices are able to provide temperature
readings or accept target temperature settings with precision equal or
better than 0.1°C. Yet the ZHA climate entity implementation advertises
an hard-coded precision of 0.5°C.

Avoid arbitrary coarsening by advertising 0.1°C precision, as passing
through the full 0.01°C precision is probably not that useful.

atmurray committed:
Bump acmeda dependency aiopulse to 0.4.2 (#43217)

adamkrol93 committed:
Fix ConnectTimeout during wolflink start (#43418)

timmo001 committed:
Add reauth support for OVO Energy (#38882)

frenck committed:
Upgrade debugpy to 1.2.0 (#43328)

frenck committed:
Add repeat mode support to Spotify (#43247)

FlavorFx committed:
Fix Luftdaten.info data retrieval (#43471)

Fix Luftdaten.info data retrieval

iprak committed:
Support for multiple states in history_stats (#43416)

Co-authored-by: Indu Prakash <6459774+InduPrakash@users.noreply.github.com>

LordBoos committed:
Upgrade discord.py to 1.5.1 (#43473)

* Upgrade discord.py to 1.5.1

discord.py 1.5.1 introduces support for Intents that are required by future bots and integrations. While this is not yet required by Home Assistant Discord integration it is needed in the future, because old API which discord.py 1.4.1 uses is now deprecated. Also it would solve dependency collision with my custom component discord_game.

* Upgrade discord.py to 1.5.1

shenxn committed:
Bump rpi-bad-power to 0.1.0 (#43476)

frenck committed:
Ensure Plex content_id in play_on_sonos service is a string (#43483)

Ernst79 committed:
Deprecate YAML config for Solar-log (#43484)

Per ADR-0010, this PR deprecates YAML configuration for Solar-log. Users who already use the Solar-log integration do not need to take action, as their configuration has already been imported into the UI.

CrashWorksLLC committed:
Add updated British Voices (#43496)

Say hello to Charlotte and James.
https://cloud.ibm.com/apidocs/text-to-speech?code=python#getvoice

OnFreund committed:
Gracefully handle no uuid in kodi discovery (#43494)

homeassistant committed:
[ci skip] Translation update

frenck committed:
Clean up Solar-log review comments (#43503)

ahertz committed:
Bump sleepyq to 0.8.1 (#43505)

amelchio committed:
Optimize Sonos queue position (#43514)

bachya committed:
Re-organize OpenUV constants (#43453)

Co-authored-by: Franck Nijhof <git@frenck.dev>

bachya committed:
Fix unhandled exception when IQVIA API fails to return data (#43359)

gtdiehl committed:
Bump envoy version to 0.17.0 (#43498)

amelchio committed:
Optimize Sonos favorites updates (#43516)

shocklateboy92 committed:
Add seek support to plex media players (#43420)

Turns out plexapi lib already supports this, so we just need to
pass through the command.

bachya committed:
Clean up RainMachine config entry (#43508)

danielrheinbay committed:
Add vendor effects to Yeelight integration (#42711)

Add effects shipped by Yeelight in their apps, as these effects are quite popular and have been missing from Home Assistant.

amelchio committed:
Optimize Sonos current playing state (#43517)

frenck committed:
Upgrade Docker base image to 2020.11.1 (#43538)

bieniu committed:
Make Brother uptime sensor disabled by default (#43478)

homeassistant committed:
[ci skip] Translation update

bachya committed:
Fix bug related to possibly missing task ID in Notion API data (#43330)

* Fix bug related to possibly missing task ID in Notion API data

* Calculate unique ID once

* Code review

* Simplify

* Code review

Kane610 committed:
Add device id to deconz_event (#43552)

dependabot[bot] committed:
Bump actions/stale from v3.0.13 to v3.0.14 (#43551)

Bumps [actions/stale](https://github.com/actions/stale) from v3.0.13 to v3.0.14.
- [Release notes](https://github.com/actions/stale/releases)
- [Commits](https://github.com/actions/stale/compare/v3.0.13...87c2b794b9b47a9bec68ae03c01aeb572ffebdb1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fronzbot committed:
Update Blinkpy to fix non-updating blink mini cameras (#43549)

amelchio committed:
Optimize reading of Sonos source mode (#43541)

Danielhiversen committed:
Mill Heater: Add support for Energy consumption (#43523)

Kane610 committed:
Track deCONZ lib changes to light based devices (#43366)

* Improve control of covers

* Log backtrace if available

* Do not create entity for controller tool
Binary sensor should use state rather than is_tripped
Add some more tests to lights and sensors

* Bump dependency to v74

* Fix Balloobs comments

frenck committed:
Upgrade Docker base image to 2020.11.2 (#43560)

squishykid committed:
update solax to 0.2.5 (#43564)

frenck committed:
Upgrade sentry-sdk to 0.19.4 (#43504)

scarface-4711 committed:
Update denonavr to 0.9.7 (#43546)

tetienne committed:
Fix RTS cover with set position available (#43555)

gtdiehl committed:
Add codeowner to Enphase Envoy manifest (#43402)

frenck committed:
Add webhook to default config (#43521)

denics committed:
Cannot use vcgencmd in HassOS (#42710)

starkillerOG committed:
Add Motion Blinds integration (#42989)

Co-authored-by: J. Nick Koston <nick@koston.org>

cereal2nd committed:
Use references in config flow for solaredge (#43511)

frenck committed:
Bump version to 0.119.0dev0 (#43485)

homeassistant committed:
[ci skip] Translation update

JJdeVries committed:
Decrease asuswrt connects per sensor (#43383)

* asuswrt: Decrease number of connects per sensor

* Changed implementation to DataUpdateCoordinator

* review changes

* update aioasuswrt dependency

* Apply suggestions from code review

Add explicit return None and fix type hint

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Fix warning generated by surpetcare test (#43598)

Kane610 committed:
Fix conversion of cover position between HASS and deCONZ (#43602)

mtdcr commented:
@weissm, could you please rebase your branch?

allenporter committed:
Update nest library and switch events to async (#43583)

springstan committed:
Add unknown_authorize_url_generation to base strings for config flows (#42484)

emontnemery committed:
Bump hatasmota to 0.1.1 (#43608)

koolsb committed:
Fix Meraki API Auth (#43578)

balloob committed:
Upgrade foobot_async to 1.0.0 (#43611)

Kane610 committed:
Add tilt support to deCONZ covers (#43607)

epenet committed:
Fix duplicate check on onewire config flow (#43590)

farmio committed:
update xknx to 0.15.4 (#43536)

allenporter committed:
Add nest SDM API camera/doorbell events (#42700)

* Add nest SDM API camera/doorbell events

Events are fired  when pubsub messages are received.   When messages
are received lookup a home assistant device id from the nest
device id, so that the home assistant device id can be included
in the event payload.

* Update homeassistant/components/nest/__init__.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

* Update nest code style based on PR feedback

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

homeassistant committed:
[ci skip] Translation update

weissm committed:
Merge branch 'homeassistant/improve/cover' of github.com:weissm/core into homeassistant/improve/cover

rytilahti committed:
Bump pyhs100 dependency to fix hs220 discoverability issues (#43619)

Fixes #39395

weissm commented:
Hallo Andreas,

 

done – let me know, if you need further changes.

 

Cheers,

Matthias

 

From: Andreas Oberritter <notifications@github.com> 
Sent: Dienstag, 24. November 2020 15:22
To: home-assistant/core <core@noreply.github.com>
Cc: Matthias Weiss <matthias.h.weiss@arcor.de>; Mention <mention@noreply.github.com>
Subject: Re: [home-assistant/core] Add better name resolution for homematic channels (#42662)

 

@weissm <https://github.com/weissm> , could you please rebase your branch?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/home-assistant/core/pull/42662#issuecomment-733005155> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABAGSHEZESMNILDEW57WMOLSRO6SDANCNFSM4TFYAHMQ> .  <https://github.com/notifications/beacon/ABAGSHGF7TPJA4NI7JPWFJDSRO6SDA5CNFSM4TFYAHM2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOFOYMKYY.gif>

pattyland committed:
Bump avea to 1.5.1 (#43618)

balloob committed:
Bump ha-ffmpeg to 3.0.2 (#43597)

pvizeli commented:
Your branch is corrupt. Please create a new branch and PR

balloob committed:
Add default config if not there (#43321)

Co-authored-by: Bram Kragten <mail@bramkragten.nl>

thecode committed:
Support for Shelly Input Events (#43479)

chemelli74 committed:
Add Shelly UNI ADC sensor (#43490)

balloob committed:
Tweak template digit detection (#43621)

cyberjunky committed:
Add FireServiceRota/BrandweerRooster integration (#38206)

Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

elupus committed:
Fix flapping derivative tests where time would move between state changes (#43579)

emontnemery committed:
Add Tasmota fan (#43353)

* Add Tasmota fan

* Bump hatasmota to 0.1.0

* Apply suggestions from code review

emontnemery committed:
Add Tasmota cover (#43368)

* Add Tasmota cover

* Update tests

* Bump hatasmota to 0.1.0

bramkragten committed:
Add more selectors (#43639)

bramkragten committed:
Add default to inputs (#43636)

ofalvai committed:
Add location to summary sensor attributes (#43641)

homeassistant committed:
[ci skip] Translation update

weissm committed:
Merge branch 'homeassistant/improve/cover' of github.com:weissm/core into homeassistant/improve/cover

homeassistant commented:
Hi @dependabot[bot], 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#42662).

Once you do that we will be able to review and accept this pull request.

Thanks!

weissm committed:
Merge branch 'homeassistant/improve/cover' of github.com:weissm/core into homeassistant/improve/cover

homeassistant commented:
Hi @dependabot[bot], 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#42662).

Once you do that we will be able to review and accept this pull request.

Thanks!

pvizeli commented:
I need create a new Branch and new PR - This branch is corrupt. Please read more about git: https://developers.home-assistant.io/docs/development_index
